crates/dslab-mp/src/mc/events.rs

```diff
@@ -1,50 +1,103 @@
-use ordered_float::OrderedFloat;
+//! Implementation of events used in model checking.
+
 use serde::Serialize;
 
+use crate::logger::LogEntry;
 use crate::message::Message;
 
-#[derive(Serialize, Clone, PartialEq, Eq, Hash, Debug)]
-pub enum DeliveryOptions {
-    /// Message will be received exactly once without corruption with specified max delay
-    NoFailures(McTime),
-    /// Message will not be received
-    Dropped,
-    /// Message delivery may be subject to some failures
-    PossibleFailures {
-        can_be_dropped: bool,
-        max_dupl_count: u32,
-        can_be_corrupted: bool,
-    },
-}
+use crate::mc::network::DeliveryOptions;
+use crate::mc::system::McTime;
 
-pub type McTime = OrderedFloat<f64>;
+/// Identifier of McEvent.
 pub type McEventId = usize;
 
+/// Special events used in model checking instead of standard events.
 #[derive(Serialize, Clone, Eq, Hash, PartialEq, Debug)]
 pub enum McEvent {
+    /// The event of receiving a non-local message by a process.
     MessageReceived {
+        /// The message itself.
         msg: Message,
+
+        /// The name of the process that sent the message.
         src: String,
+
+        /// The name of the process that received the message.
         dest: String,
+
+        /// Network delivery options for the message.
         options: DeliveryOptions,
     },
+
+    /// The event of a timer expiration.
     TimerFired {
+        /// The process to which the timer belongs to.
         proc: String,
+
+        /// The timer name.
         timer: String,
+
+        /// The timer duration.
         timer_delay: McTime,
     },
+
+    /// The event of cancelling a timer.
     TimerCancelled {
+        /// The process to which the timer belongs to.
         proc: String,
+
+        /// The timer name.
         timer: String,
     },
+
+    /// The event of dropping a message. Created by the model checking strategy.
     MessageDropped {
+        /// The dropped message itself.
+        msg: Message,
+
+        /// The name of the process that sent the message.
+        src: String,
+
+        /// The name of the process the message was sent to.
+        dest: String,
+    },
+
+    /// The event of duplicating a message. Created by model checking strategy.
+    MessageDuplicated {
+        /// The duplicated message itself.
+        msg: Message,
+
+        /// The original message id.
+        msg_id: McEventId,
```

Comment:
Предалагаю поправить так везде (см. мою ветку `mc-trace-sol`)

```suggestion
        /// The id of original MessageReceived event.
        receive_event_id: McEventId,
```